controllers/binding/snapshotenvironmentbinding_controller.go

```diff
@@ -199,6 +199,6 @@ func SetupController(manager ctrl.Manager, log *logr.Logger) error {
 // setupControllerWithManager sets up the controller with the Manager which monitors new SnapshotEnvironmentBindings
 func setupControllerWithManager(manager ctrl.Manager, reconciler *Reconciler) error {
 	return ctrl.NewControllerManagedBy(manager).
-		For(&applicationapiv1alpha1.SnapshotEnvironmentBinding{}, builder.WithPredicates(predicates.GenerationUnchangedOnUpdatePredicate{}, gitops.DeploymentSucceededForIntegrationBindingPredicate())).
+		For(&applicationapiv1alpha1.SnapshotEnvironmentBinding{}).WithEventFilter(predicate.Or(predicates.GenerationUnchangedOnUpdatePredicate{}, gitops.DeploymentSucceededForIntegrationBindingPredicate(), gitops.DeploymentFailedForIntegrationBindingPredicate())).
```

Comment:
I think that the `predicates.GenerationUnchangedOnUpdatePredicate{}` shouldn't be part of the `Or` clause here because that would then catch all update events to the resource and make the other two predicates redundant if I'm not mistaken. @hongweiliu17 can you also take a look at this?

Reply:
Yes. I found this [Or](https://github.com/onsi/gomega/blob/master/matchers.go#L609-L617)